src/lib/es5.d.ts

```diff
@@ -1708,7 +1704,7 @@ interface Int8Array {
       * @param start The beginning of the specified portion of the array.
       * @param end The end of the specified portion of the array.
       */
-    slice(start?: number, end?: number): Int8Array;
+    slice(start?: number, end?: number): T;

```

Comment:
Rather than return `T`, either return `this` or return `TypedArray` with a subtyped overload in each specific typed-array constructor (e.g. return `Int32Array` in the `Int32Array` interface).